src/redux/books/booksSlice.js

```diff
@@ -1,46 +1,86 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const baseUrl = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/uRddr6OSr0tnEmWTsb26/books';
 
 const initialState = {
-  books:
-  [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: false,
+  error: undefined,
+
 };
 
-const booksSlice = createSlice({
+export const getBooks = createAsyncThunk('book/get', async () => {
+  try {
+    const response = await axios.get(baseUrl);
+    return response.data;
+  } catch (err) {
+    return err.message;
+  }
+});
```

Comment:
- [ ] I also noticed that even though you created this action for fetching books from the API, you didn't implement it in your `BooksDisplay` component.
- [ ] However per the project requirement, you need to ensure that `fetchBooks` or in your case `getBooks` actually updates the state of `booksSlice` 😉 
- [ ] You also need to ensure that `useEffect` is used when dispatching the `fetchBooks` function in `<BookList>` or in your case the `<BooksDisplay>` component 😉 
- [ ] This way, when a new book is added to the API, it will be fetched from the API and displayed on the page so when you refresh the page, any books that haven't been deleted from the API will still be visible 💯 
- [ ] Currently in your app, the moment the page is refreshed, any books that were added disappear from the page this is because the aren't being gotten from the API properly so kindly work on this 👍 
- [ ] Lastly, there should be two input options in the form for adding a book. `Book title `and `Author` but from the video below currently you have only one so kindly add the second one as well 😉 
- Video Reference

https://github.com/Bellagirl-maker/book-store/assets/102699958/3ba9e675-f478-42b3-8c44-b1179f91370a

Reply:
@NahnahAJ Done with these changes